packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -375,6 +374,10 @@ export class ForkChoice implements IForkChoice {
       justifiedRoot: toHexString(state.currentJustifiedCheckpoint.root),
       finalizedEpoch: finalizedCheckpoint.epoch,
       finalizedRoot: toHexString(state.finalizedCheckpoint.root),
+
+      // Optimistic sync based status to verify/prune later
+      payloadStatusUnknown: preCachedData?.payloadStatusUnknown,
+      isMergeBlock: preCachedData?.isMergeBlock,
```

Comment:
Is it really necessary to track `isMergeBlock`? ProtoBlocks already track executionPayloadBlockHash. Would that be enough?

Reply:
Checked Lighthouse's implementation and they only use
```rs
execution_status: ExecutionStatus
```

https://github.com/sigp/lighthouse/pull/2686/files#diff-446884cdc88e364a89436e4af6ef3b563c32556d6900679c9117bf50e93b3b72R40

Replacing existing `executionPayloadBlockHash` field

Reply:
if we wish to enumerate the `payloadStatus` field, these two fields can be combined in a structure: 
enum ('VALID','INVALID','UNKNOWN','PREMERGE','MERGEVALIDPOW','MERGEUNKNOWNPOW')

merge block need special tracking to run the `assertValidTerminalPowBlock` / TTD difficulty validation if it was skipped because of EL syncing.
we can use `executionPayloadBlockHash` for merge block tracking as its parent will have that as null (so we can skip `MERGEVALIDPOW`, `MERGEUNKNOWNPOW` but then will always need to check the parent of the first validation for merge ttd checks)

another Q: do we need to track 'INVALID' status as they would anyway be pruned? may be it will help on rethrowing critical error on restarts if the finalized/justified checkpoint block becomes invalid

Reply:
Right let's do
```ts
enum ExecutionStatus {
  Valid,
  Syncing,
  PreMerge
}

type IProtoBlock = 
  { executionStatus: ExecutionStatus.Valid | ExecutionStatus.Syncing, executionPayloadBlockHash: RootHex }
  | { executionStatus: ExecutionStatus.PreMerge, executionPayloadBlockHash: null }
```